backend/ccm/canvas_api/admin_sections_api_handler.py

```diff
@@ -0,0 +1,189 @@
+        term_id = request.query_params.get('term_id', None)
+        instructor_name = request.query_params.get('instructor_name', None)
+        course_name = request.query_params.get('course_name', None)
+
+        if not term_id:
+            return Response({"error": "Term ID is required as a parameter"}, status=HTTPStatus.BAD_REQUEST)
+        
+        if not ((instructor_name != None) ^ (course_name != None)): # XOR condition ensures one, but not both, is provided
+            return Response({"error": "Provide either 'instructor_name' or 'course_name' as a parameter. Both cannot be provided together."},
+                            status=HTTPStatus.BAD_REQUEST)
+        time_get_canvas_api_start: float = time.perf_counter()
+        canvas_api: Canvas = await asyncio.to_thread(self.credential_manager.get_canvasapi_instance, request)
+        time_get_canvas_api_end: float = time.perf_counter()
+        logger.info(f"TIME taken to get Canvas API instance: {time_get_canvas_api_end - time_get_canvas_api_start:.2f} seconds")
+        
+        logger.info(f"Retrieving admin sections data for term_id: {term_id}")
+        time_filter_accounts_start: float = time.perf_counter()
+        try:
+            # Get accounts accessible to the admin user
+            accounts_by_account_id = await self.retrieve_accounts_map(canvas_api)
+            filtered_account_ids = [
+                account.id for account in accounts_by_account_id.values() if
+                    # account is the root account OR the subaccount of an unlisted account
+                    account.parent_account_id is None 
+                    or not(account.parent_account_id in accounts_by_account_id.keys())
+            ]
+            logger.info(f"Found {len(filtered_account_ids)} of {len(accounts_by_account_id)} accounts accessible to the admin user.")
+        except (CanvasException, Exception) as e:
+            failed_input = f"user {request.user}"
+            self.canvas_error.handle_canvas_api_exceptions(HTTPAPIError(failed_input, e))
+            return Response(self.canvas_error.to_dict(), status=self.canvas_error.to_dict().get('statusCode'))
+        time_filter_accounts_end: float = time.perf_counter()
+        logger.info(f"TIME taken to filter accounts: {time_filter_accounts_end - time_filter_accounts_start:.2f} seconds")
+        
+        # Build term and search parameters for course search
+        courseQueryParams = {
+            'state': ['created', 'claimed', 'available'],
+            'enrollment_term_id': term_id,
+            'per_page': 100,
+        }
+        if instructor_name:
+            courseQueryParams['by_teachers'] = ['sis_login_id:' + instructor_name]
+            logger.info(f"Searching for courses with instructor name: {instructor_name}")
+        if course_name:
+            courseQueryParams['search_term'] = course_name
+            logger.info(f"Searching for courses with name: {course_name}")
```

Comment:
I feel L85-90 could be written before the accounts API call if the request by uniqname or course_name. That will be a better code flow IMHO.